src/strings_type.h

```diff
@@ -10,6 +10,7 @@
 #ifndef STRINGS_TYPE_H
 #define STRINGS_TYPE_H
 
+#include "core/overflowsafe_type.hpp"
```

Comment:
I don't see any overflow safe type being used.
```suggestion
#include "core/convertible_through_base.hpp"
```